### PR TITLE
[String] Add TruncateMode mode to truncate methods

### PR DESCRIPTION
**Pull request by Timherlaud**
update documentation for related  issue : #20020

### CODE REVIEW

File: string.rst [hunk 1]
```diff
@@ -394,11 +394,16 @@ Methods to Join, Split, Truncate and Reverse
...
     // the second argument is the character(s) added when a string is cut
+    // the third argument is TruncateMode::Char by default
     // (the total length includes the length of this character(s))
-    u('Lorem Ipsum')->truncate(8, '…');             // 'Lorem I…'
-    // if the third argument is false, the last word before the cut is kept
-    // even if that generates a string longer than the desired length
-    u('Lorem Ipsum')->truncate(8, '…', cut: false); // 'Lorem Ipsum'
+    u('Lorem Ipsum')->truncate(8, '…');        // 'Lorem I…'
+    // use options to keep complete words
+    u('Lorem ipsum dolor sit amet')->truncate(10, '...', TruncateMode::WordBefore);  // 'Lorem...'
```
Comment (94noni) [inline string.rst:401]: I would use the example of the original PR :)
appart from that, nice addition 👍🏻

**Comment by javiereguiluz** [comment]
Thanks Tim!

File: string.rst (hunk 1 above)
Comment (javiereguiluz) [reply string.rst:401]: Yes, I did some minor changes while merging. Thanks!

File: string.rst [hunk 2]
```diff
@@ -394,11 +394,16 @@ Methods to Join, Split, Truncate and Reverse
...
+    u('Lorem ipsum dolor sit amet')->truncate(14, '...', TruncateMode::WordAfter);  // 'Lorem ipsum...'
+
+.. versionadded:: 7.2
+
+    The TruncateMode argument for truncate function was introduced in Symfony 7.2.
```
Comment (OskarStark) [inline string.rst:406]:
```suggestion
    The TruncateMode parameter for truncate function was introduced in Symfony 7.2.
```

Comment (OskarStark) [reply string.rst:406]: Fixed by myself

**Pull request merged**